packages/cardpay-sdk/sdk/reward-manager/base.ts

```diff
@@ -308,6 +315,143 @@ export default class RewardManager {
+      throw new Error('prepaidCardAddress is required');
+    }
+    if (!rewardProgramId) {
+      throw new Error('rewardProgramId is required');
+    }
+    if (!blob) {
+      throw new Error('blob is required');
+    }
+    let prepaidCardAddress = prepaidCardAddressOrTxnHash;
+    let { nonce, onNonce, onTxnHash } = txnOptions ?? {};
+    let from = contractOptions?.from ?? (await this.layer2Web3.eth.getAccounts())[0];
+
+    await this.checkPrepaidCardOwnerIsAdmin(rewardProgramId, prepaidCardAddress);
+
+    let gnosisResult = await executeSendWithRateLock(this.layer2Web3, prepaidCardAddress, async (rateLock) => {
+      let payload = await this.getAddRewardRulePayload(prepaidCardAddress, rewardProgramId, blob, rateLock);
+      if (nonce == null) {
+        nonce = getNextNonceFromEstimate(payload);
+        if (typeof onNonce === 'function') {
+          onNonce(nonce);
+        }
+      }
+      return await this.executeAddRewardRule(
+        prepaidCardAddress,
+        rewardProgramId,
+        blob,
+        rateLock,
+        payload,
+        await signPrepaidCardSendTx(this.layer2Web3, prepaidCardAddress, payload, nonce, from),
+        nonce
+      );
+    });
+
+    if (!gnosisResult) {
+      throw new Error(
+        `Unable to obtain a gnosis transaction result for lock reward program from prepaid card ${prepaidCardAddress}`
+      );
+    }
+
+    let txnHash = gnosisResult.ethereumTx.txHash;
+
+    if (typeof onTxnHash === 'function') {
+      await onTxnHash(txnHash);
+    }
+    return await waitForSubgraphIndexWithTxnReceipt(this.layer2Web3, txnHash);
+  }
+
+  async removeRewardProgram(txnHash: string): Promise<TransactionReceipt>;
+  async removeRewardProgram(
+    safeAddressIdOrTxnHash: string,
+    rewardProgramId: string,
+    txnOptions?: TransactionOptions,
+    contractOptions?: ContractOptions
+  ): Promise<TransactionReceipt>;
+  async removeRewardProgram(
+    safeAddressIdOrTxnHash: string,
+    rewardProgramId?: string,
+    txnOptions?: TransactionOptions,
+    contractOptions?: ContractOptions
+  ): Promise<TransactionReceipt> {
+    if (isTransactionHash(safeAddressIdOrTxnHash)) {
+      let txnHash = safeAddressIdOrTxnHash;
+      return await waitForSubgraphIndexWithTxnReceipt(this.layer2Web3, txnHash);
+    }
+    if (!safeAddressIdOrTxnHash) {
+      throw new Error('safeAddress is required');
+    }
+    if (!rewardProgramId) {
+      throw new Error('rewardProgramId is required');
+    }
+    let { nonce, onNonce, onTxnHash } = txnOptions ?? {};
+    let safeAddress = safeAddressIdOrTxnHash;
+    let from = contractOptions?.from ?? (await this.layer2Web3.eth.getAccounts())[0];
+    let rewardManagerAddress = await getAddress('rewardManager', this.layer2Web3);
+    let rewardManager = await this.getRewardManager();
+
+    let payload = rewardManager.methods.removeRewardProgram(rewardProgramId).encodeABI();
+    let gasTokenAddress = await this.defaultGasToken();
```

Comment:
what? there is no "default gas token". I thought his was just going to be an EOA txn, which could optionally be a DAO in the future.

Reply:
it might even be the case where this is executed via the gnosis hosted relay server and not our relay server in that future scenario.

Reply:
crap--yeah that unsupported chain ID i think runs deep. I've seen it awhile back looking at an unrelated thing. but there is like a dep of a dep of a dep that has the network ID's hardcoded into it. because we don't do EOA txn's in layer 2 we normally woudn't see this.

So for EOA txn's it's totally ok to assume that the address has access to native coin--since that is the way ethereum actually works. This particular function is actually a function that I think that we as cardstack would only ever run (for the time being). Moreover, we would probably use blockscout to run this and not the SDK, and we defnite do have access to native coin. I think the answer here is to just remove this from the SDK. This is not a function that I would anticipate any of our current protocol clients (dap, cli, wallet app) would actually call.

Reply:
>  I think the answer here is to just remove this from the SDK. This is not a function that I would anticipate any of our current protocol clients (dap, cli, wallet app) would actually call.

Makes lots of sense. I will remove it from the sdk.

Reply:
@habdelra I removed the `removeRewardProgram`. Good to merge?